ttnn/cpp/ttnn/tensor/tensor_ops.cpp

```diff
@@ -402,10 +407,16 @@ Tensor tensor_reshape(const Tensor& input_tensor, const ttnn::Shape& new_shape)
                         auto shard_spec = shard_spec_buffer.tensor_shard_spec;
                         auto shard_shape = shard_spec.shape;
 
-                        uint32_t mul_div = new_shape[-1] > shard_shape[1] ? (new_shape[-1] / shard_shape[1])
```

Comment:
No need to calculate it all. Please just ask tensor spec.
```
const auto& tensor_spec = tensor.tensor_spec();
auto shard_spec_buffer = tensor_spec.compute_shard_spec_buffer();
device_buffer-> set_shard_spec(shard_spec_buffer);
```

Reply:
@TT-BrianLiu , look how handy :)

Reply:
I try to replaced with `compute_shard_spec_buffer`, but the `test_reshape_sharded_rm` failed@@
```
# origin
page_shape is {1, 32}
tensor2d_shape is {128, 1}
# use compute_shard_spec_buffer
page_shape is {1, 64}
tensor2d_shape is {4096, 1}

E       RuntimeError: TT_FATAL @ /home/ubuntu/tt-metal/ttnn/cpp/ttnn/operations/data_movement/transpose/device/transpose_op.cpp:57: shard_spec.shape[1] == W
```

so I kept the origin code